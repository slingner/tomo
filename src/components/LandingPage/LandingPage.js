import React from 'react';
import {
  PageWrapper,
  LandingHeader,
  ContentHeader,
  LandingText,
  LandingLink,
  ContentSection,
} from './LandingPage.style';
import { FormButton } from '../Button/Button';
import { FormInput, FormWrapper } from '../Form/Form';
import { colors } from '../../constants';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export default function LandingPage() {
  return (
    <PageWrapper>
      <ContentHeader>
        <LandingHeader> Credit for all.</LandingHeader>
        <FormWrapper>
          <FormInput
            id='Email'
            name='Email'
            aria-label='Enter your email here.'
            aria-required='true'
            required
            color={colors.darkgrey}
            placeholder='Enter Your Email...'
          />
          <FormButton>Sign Up!</FormButton>
        </FormWrapper>
      </ContentHeader>
      <ContentSection
        style={{
          display: 'flex',
          flexDirection: 'row',
          backgroundColor: colors.darkpurple,
        }}
      >
        <FontAwesomeIcon icon='tv' />
        <FontAwesomeIcon icon='check-circle' />
        <FontAwesomeIcon icon='layer-group' />
      </ContentSection>
      <ContentSection style={{ backgroundColor: colors.darkblue }}>
        Hello
      </ContentSection>
      <ContentSection style={{ backgroundColor: colors.darkgrey }}>
        Hello
      </ContentSection>
    </PageWrapper>
  );
}
