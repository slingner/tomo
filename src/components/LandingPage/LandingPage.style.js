import styled from 'styled-components';
import { colors } from '../../constants';
import wallet from './assets/wallet.jpg';

export const PageWrapper = styled.div`
  text-align: center;
  width: 100vw;
`;

export const LandingHeader = styled.h1`
  text-align: center;
  padding: 50px 10px 30px 10px;
  color: ${colors.white};
  font-size: 65px;
  width: 100vw;
  opacity: 100%;

  @media (min-width: 600px) {
    padding-top: 350px;
  }
`;

export const LandingText = styled.p`
  text-align: center;
  color: ${colors.white};
  font-size: 25px;
  padding: 35px;
  margin: auto;

  @media (min-width: 00px) {
  }
`;

export const ContentHeader = styled.div`
  background-image: url(${wallet});
  height: 100%;
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  .header {
    background-color: rgba(66, 75, 84, 0.5);
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    height: 100%;
    z-index: 1px;
  }
  @media (min-width: 600px) {
    height: 100vh;
  }
`;

export const ContentSection = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: fit-content;
  color: ${colors.white};
  background-color: ${(props) =>
    props.color ? props.color : colors.slategrey};
  align-items: center;
  font-size: 20px;
  text-align: center;
  padding: 100px;

  @media (min-width: 800px) {
    flex-direction: row;
  }
`;

export const LandingImage = styled.img`
  margin-left: auto;
  margin-right: auto;
  border: 1px solid grey;
  border-radius: 5px;
  height: 100%;
  height: auto;

  @media (min-width: 600px) {
    width: 70%;
    height: auto;
  }
`;
