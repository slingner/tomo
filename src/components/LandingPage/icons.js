import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import styled, { keyframes } from 'styled-components';
import cx from 'classnames';
import { pulse } from 'react-animations';

export const Credit = React.forwardRef(({ className, ...props }, ref) => {
  return (
    <div className={cx('Div', className)} ref={ref} {...props}>
      <FontAwesomeIcon icon='credit-card' />
    </div>
  );
});

const bounceAnimation = keyframes`${pulse}`;

export const CreditBounce = styled(Credit)`
  background: transparent;
  border: none;
  font-size: 60px;

  padding: 20px;

  :hover {
    animation: 1s ${bounceAnimation};
  }
  @media (min-width: 800px) {
    font-size: 120px;
  }
`;

export const Invoice = React.forwardRef(({ className, ...props }, ref) => {
  return (
    <div className={cx('Div', className)} ref={ref} {...props}>
      <FontAwesomeIcon icon='file-invoice-dollar' />
    </div>
  );
});

export const InvoiceBounce = styled(Invoice)`
  background: transparent;
  border: none;
  font-size: 60px;
  padding: 20px;

  :hover {
    animation: 1s ${bounceAnimation};
  }

  @media (min-width: 800px) {
    font-size: 120px;
  }
`;

export const Bill = React.forwardRef(({ className, ...props }, ref) => {
  return (
    <div className={cx('Div', className)} ref={ref} {...props}>
      <FontAwesomeIcon icon='money-bill-wave' />
    </div>
  );
});

export const BillBounce = styled(Bill)`
  background: transparent;
  border: none;
  font-size: 60px;
  padding: 20px;

  :hover {
    animation: 1s ${bounceAnimation};
  }

  @media (min-width: 800px) {
    font-size: 120px;
  }
`;
