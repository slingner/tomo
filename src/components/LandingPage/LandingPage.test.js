import React from "react";
import { shallow, configure } from "enzyme";
import toJson from "enzyme-to-json";
import LandingPage from "./LandingPage";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });
describe(`<LandingPage />`, () => {
  //Snapshot Testing
  it("renders LandingPage by default", () => {
    const wrapper = shallow(<LandingPage />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
