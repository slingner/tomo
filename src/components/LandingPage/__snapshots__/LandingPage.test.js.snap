// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPage /> renders LandingPage by default 1`] = `
<styled.div>
  <styled.div>
    <styled.h1>
      Start with no credit. 
      <br />
      Build amazing credit.
    </styled.h1>
    <styled.form
      onSubmit={[Function]}
    >
      <Styled(Component)
        color="#0D0D0D"
        name="email"
        onChange={[Function]}
        placeholder="Enter Your Email..."
        required={true}
        type="email"
        value=""
      />
      <Styled(Component)
        type="submit"
      >
        Sign Up!
      </Styled(Component)>
      <CSSTransition
        appear={true}
        classNames="alert"
        in={false}
        timeout={100}
        unmountOnExit={true}
      >
        <p
          style={
            Object {
              "color": "white",
            }
          }
        >
          Success!
        </p>
      </CSSTransition>
    </styled.form>
    <div
      style={
        Object {
          "borderRadius": "5px",
          "color": "white",
          "fontSize": "22px",
          "margin": "auto",
          "maxHeight": "200px",
          "maxWidth": "300px",
          "minHeight": "150px",
          "overflow": "auto",
        }
      }
    >
      All emails: 
       
    </div>
  </styled.div>
  <styled.div
    style={
      Object {
        "backgroundColor": "#BF5A1F",
      }
    }
  >
    <styled.div>
      <Styled(Component) />
      <styled.p>
        No credit score? Np.
      </styled.p>
    </styled.div>
    <styled.div>
      <Styled(Component) />
      <styled.p>
        Never miss a payment.
      </styled.p>
    </styled.div>
    <styled.div>
      <Styled(Component) />
      <styled.p>
         Earn cashback.
      </styled.p>
    </styled.div>
  </styled.div>
  <styled.div
    style={
      Object {
        "backgroundColor": "#F2A03D",
      }
    }
  >
    <styled.p>
      <styled.span>
        Build credit the smart way. 
      </styled.span>
      <br />
      <br />
      Tomo is the first card to offer specialized smart payments catered to credit building. Choose to pay off your card weekly, bi-weekly or monthly.
    </styled.p>
    <styled.img
      alt="money"
      src="cash.jpg"
    />
  </styled.div>
  <styled.div
    style={
      Object {
        "backgroundColor": "#F2CC85",
      }
    }
  >
    <styled.p
      style={
        Object {
          "padding": "30px",
        }
      }
    >
      Not only do you automatically earn 1% cash back by default, but have the ability to boost your cashback up to 20%.
      <br />
      <br />
      <styled.span>
        Yep, pretty amazing we know.
      </styled.span>
    </styled.p>
    <styled.img
      alt="smiles"
      src="happy.jpg"
    />
  </styled.div>
  <styled.div>
    <styled.p>
      You work hard for your money and simply put,
      <br />
      <br />
      <styled.span>
        credit card fees suck.
      </styled.span>
      <br />
      <br />
      That’s why we got rid of them and don’t plan on having them, ever.
    </styled.p>
    <styled.img
      alt="money"
      src="balloon.jpg"
    />
  </styled.div>
  <styled.h1
    style={
      Object {
        "color": "black",
      }
    }
  >
    What people are saying...
  </styled.h1>
  <styled.div>
    <styled.div>
      <styled.img
        alt="Portrait of guy"
        src="guy.jpg"
      />
      <styled.span
        style={
          Object {
            "color": "black",
          }
        }
      >
        David W. 
      </styled.span>
      <styled.p
        style={
          Object {
            "color": "#404040",
          }
        }
      >
        "This is fantastic! Thanks so much guys!"
      </styled.p>
    </styled.div>
    <styled.div>
      <styled.img
        alt="Portrait of another guy"
        src="guy2.jpg"
      />
      <styled.span
        style={
          Object {
            "color": "black",
          }
        }
      >
        Matt L. 
      </styled.span>
      <styled.p
        style={
          Object {
            "color": "#404040",
          }
        }
      >
        "Tomo is amazing. I've been using it to build my credit."
      </styled.p>
    </styled.div>
    <styled.div>
      <styled.img
        alt="Portrait of girl"
        src="girl2.jpg"
      />
      <styled.span
        style={
          Object {
            "color": "black",
          }
        }
      >
        Tammy T. 
      </styled.span>
      <styled.p
        style={
          Object {
            "color": "#404040",
          }
        }
      >
        "Thanks so much for making these resources available to us!"
      </styled.p>
    </styled.div>
  </styled.div>
</styled.div>
`;
